draft-rescorla-tls-esni.md

```diff
@@ -365,6 +369,66 @@ has cleaned out most such proxies.
 
 ## Comparison Against Criteria
 
+{{?I-D.ietf-tls-sni-encryption}} lists several requirements for SNI encryption. In this 
+section, we re-iterate these requirements and assess the ESNI design against them.
+
+1. Mitigate against replay attacks
+
+Since K_sni is derived from a (EC)DH operation between the client's ephemeral
+and server's semi-static ESNI key. This binds the ESNI encryption to the 
+Client Hello. It is not possible for an attacker to "cut and paste" the ESNI
+value in a different Client Hello, with a different ephemeral key share, as
+the terminating server will fail to decrypt and verify the ESNI value.
+
+2. Avoid widely-deployed shared secrets
+
+This design depends upon DNS as a vehicle for semi-static public key distribution.
+Server operators may partition their private keys however they see fit provided
+each server behind an IP address has the corresponding private key to decrypt 
+a key. Thus, when one ESNI key is provided, sharing is optimally bound by the number 
+of hosts that share an IP address. Server operators may further limit sharing
+by including multiple keys, with distinct labels, in an ESNIKeys structure.
+
+3. Prevent SNI-based DoS attacks
+
+This design requires servers to decrypt ClientHello messages with EncryptedSNI
+extensions carrying valid labels. Thus, it is possible for an attacker to force
+decryption operations on the server. Servers SHOULD NOT be configured to handle
+multiple keys, as this then require servers to perform trial decryption before
+learning of the attack. This attack is bound by the number of valid TCP connections
+an attacker can open. 
+
+4. Do not stick out
+
+By sending SNI and ESNI values (with illegitimate labels), or by sending 
+legitimate ESNI values for and "fake" SNI values, clients do not display
+clear signals of ESNI intent to passive eavesdroppers. As more clients 
+enable ESNI support with keys supplied by shared hosting providers, the 
+presence of ESNI extensions becomes less suspicious.
```

Comment:
I think you should sharpen this point a bit. The argument I would make is that this is a property of the browser and the server, so it's not suspicious at all. Now, it's true that you need more clients to avoid *blocking* but if you're using a generic browser, there's nothing suspicious about running Safari, Firefox, etc.